keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt

```diff
@@ -52,11 +52,16 @@ internal class EnvironmentPromotionCheckerTests : JUnit5Minutests {
       every { isImplicit() } returns true
       every { canPromote(any(), any(), any(), any()) } returns true
     }
-    val subject = EnvironmentPromotionChecker(
+    val environmentConstraintRunner = EnvironmentConstraintRunner(
       repository,
       listOf(statelessEvaluator, statefulEvaluator, implicitStatelessEvaluator),
       publisher
     )
+    val subject = EnvironmentPromotionChecker(
+      repository,
+      environmentConstraintRunner,
+      publisher
+    )
 
```

Comment:
In addition to adding the new dependency here, I had to update the mocks to provide the new behavior: add to the queued table + read from that to approve. Given I had to do minimal updates, I feel pretty good. I'd feel better if these tests used sql instead of a mocks, but I'm not sure if it's a good idea to rewrite them to do that.